hatchet/readers/hpctoolkit_reader_latest.py

```diff
@@ -0,0 +1,421 @@
+            formatValues = "<Hd"
+            values_sub_db = file.read(nValues * struct.calcsize(formatValues))
+
+            for i in range(nCtxs):
+                (ctxId, startIndex) = safe_unpack("<LQ", cct_sub_db, 0, i)
+                cct_id = ctxId
+
+                if i == nCtxs - 1:
+                    end_index = nValues
+                else:
+                    (_, end_index) = safe_unpack(formatCtxs, cct_sub_db, 0, i + 1)
+
+                self._summary_profile[cct_id] = {}
+
+                for j in range(startIndex, end_index):
+                    (metricId, value) = safe_unpack("<Hd", values_sub_db, 0, j)
+
+                    if metricId in self._metric_descriptions:
+                        self._summary_profile[cct_id][
+                            self._metric_descriptions[metricId]
+                        ] = value
+
+                        if self._metric_descriptions[metricId].endswith("(i)"):
+                            self._inclusive_metrics[metricId] = (
+                                self._metric_descriptions[metricId]
+                            )
+                        else:
+                            self._exclusive_metrics[metricId] = (
+                                self._metric_descriptions[metricId]
+                            )
+
+    def _read_cct(
+        self,
+    ) -> None:
+        hatchet_roots = []
+
+        with open(self._meta_file, "rb") as file:
+            meta_db = file.read()
+
+        (pContext,) = safe_unpack("<Q", meta_db, FILE_HEADER_OFFSET + 7 * 8)
+        (pEntryPoints, nEntryPoints, szEntryPoint) = safe_unpack(
+            "<QHB", meta_db, pContext
+        )
+
+        for i in range(nEntryPoints):
+            (szChildren, pChildren, ctxId, entryPoint) = safe_unpack(
+                "<QQLH",
+                meta_db,
+                pEntryPoints,
+                i,
+                szEntryPoint,
+            )
+
+            if entryPoint != 1:
+                continue
+
+            frame = {
+                "id": ctxId,
+                "type": "entry",
+                "name": entryPoint,
+                "depth": 0,
+            }
+            entry_node = Node(Frame(frame), parent=None, hnid=ctxId)
+
+            hatchet_roots.append(entry_node)
+
+            entry_value = {"node": entry_node, "name": "entry"}
+
+            if ctxId in self._summary_profile:
+                entry_value.update(self._summary_profile[ctxId])
+
+            self._metrics_table.append(entry_value)
+
+            try:
+                my_time = self._summary_profile[ctxId][self._time_metric]
+            except:
+                my_time = None
+
+            self._total_execution_time = my_time
+
+            self._parse_context(
+                pChildren,
+                szChildren,
+                entry_node,
+                meta_db,
+                my_time
+            )
+
+            table = pd.DataFrame(self._metrics_table).set_index("node")
```

Comment:
Shouldn't this (and everything in this function below this line) be less indented? Right now, all this code for creating the `DataFrame`, `Graph`, and `GraphFrame` are inside the for loop over "entry points". If I'm understanding the code correctly, that means that, if you had more than one entry point, the function would return early and skip all but the first entry point.

Reply:
@draganagrbicllnl I agree partial unwinds are confusing/usually system calls etc. and likely unnecessary.  How does HPC Toolkit ensure that entryPoint 1 is the application?  Will this remain being the case going forward?  If so, I am ok with it.